shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/Alert.kt

```diff
@@ -205,36 +285,38 @@ data class Alert(
 
     fun anyInformedEntity(predicate: (InformedEntity) -> Boolean) = informedEntity.any(predicate)
 
+    fun anyInformedEntitySatisfies(predicateBuilder: InformedEntity.PredicateBuilder.() -> Unit) =
+        informedEntity.any { it.satisfies(predicateBuilder) }
+
     fun matchingEntities(predicate: (InformedEntity) -> Boolean) = informedEntity.filter(predicate)
 
     companion object {
         /**
-         * Returns alerts that are applicable to the passed in routes and stops
+         * Returns alerts that are applicable to the passed in routes, stops, and trips
          *
          * Criteria:
          * - Route ID matches an alert [Alert.InformedEntity]
          * - Stop ID matches an alert [Alert.InformedEntity]
          * - Alert's informed entity activities contains [Alert.InformedEntity.Activity.Board]
+         * - Trip ID matches an alert [Alert.InformedEntity]
          */
         fun applicableAlerts(
             alerts: Collection<Alert>,
             directionId: Int?,
             routeIds: List<String>,
-            stopIds: Set<String>?
+            stopIds: Set<String>?,
+            tripId: String?
         ): List<Alert> {
             return alerts
                 .filter { alert ->
-                    alert.anyInformedEntity {
-                        routeIds.any { routeId ->
-                            stopIds?.any { stopId ->
-                                it.appliesTo(
-                                    directionId = directionId,
-                                    routeId = routeId,
-                                    stopId = stopId
-                                )
-                            }
-                                ?: it.appliesTo(directionId = directionId, routeId = routeId)
-                        } && it.activities.contains(Alert.InformedEntity.Activity.Board)
+                    alert.anyInformedEntitySatisfies {
+                        checkActivity(InformedEntity.Activity.Board)
+                        checkDirection(directionId)
+                        checkRouteIn(routeIds)
+                        if (stopIds != null) {
+                            checkStopIn(stopIds)
+                        }
+                        checkTrip(tripId)
                     }
```

Comment:
It's possible that the `InformedEntity.PredicateBuilder` is too complicated, but I think this code is both clearer and more efficient (comparing the direction O(N²) times is a waste), and this isn't even why I introduced it.